SETUP/check_require_login.php

```diff
@@ -0,0 +1,186 @@
+    "quiz/generic/proof.php",
+    "quiz/generic/orig.php",
+    "quiz/generic/hints.php",
+    "quiz/generic/returnfeed.php",
+    "quiz/generic/main.php",
+    "quiz/generic/right.php",
+    "quiz/tuts/tut_p_mod2_3.php",
+    "quiz/tuts/tut_p_mod2_1.php",
+    "quiz/tuts/tut_p_mod1_4.php",
+    "quiz/tuts/tut_p_greek_2.php",
+    "quiz/tuts/tut_p_mod1_5.php",
+    "quiz/tuts/tut_p_mod1_3.php",
+    "quiz/tuts/tut_p_basic_1.php",
+    "quiz/tuts/tut_p_basic_3.php",
+    "quiz/tuts/tut_p_basic_5.php",
+    "quiz/tuts/tut_p_old_3.php",
+    "quiz/tuts/tut_p_basic_4.php",
+    "quiz/tuts/tut_p_old_2.php",
+    "quiz/tuts/tut_p_aeoe_2.php",
+    "quiz/tuts/tut_p_mod2_2.php",
+    "quiz/tuts/tut_p_mod1_2.php",
+    "quiz/tuts/tut_p_thorn.php",
+    "quiz/tuts/tut_p_aeoe_1.php",
+    "quiz/tuts/tut_p_greek_5.php",
+    "quiz/tuts/tut_p_greek_1.php",
+    "quiz/tuts/tut_p_mod2_4.php",
+    "quiz/tuts/tut_p_mod1_1.php",
+    "quiz/tuts/tut_p_basic_2.php",
+    "quiz/tuts/tut_p_old_1.php",
+    "quiz/tuts/tut_p_greek_4.php",
+    "quiz/tuts/tut_p_mod2_5.php",
+    "quiz/tuts/tut_p_fraktur.php",
+    "quiz/tuts/tut_p_greek_3.php",
+    # Stats graphs that we make globally available
+    "stats/default.php",
+    "stats/jpgraph_files/percent_users_who_proof.php",
+    "stats/jpgraph_files/pages_daily.php",
+    "stats/jpgraph_files/cumulative_total_proj_summary_graph.php",
+    "stats/jpgraph_files/users_by_pages_proofed_graph.php",
+    "stats/jpgraph_files/users_by_country.php",
+    "stats/jpgraph_files/round_backlog_trend.php",
+    "stats/jpgraph_files/round_backlog_days.php",
+    "stats/jpgraph_files/round_backlog.php",
+    "stats/jpgraph_files/cumulative_total_proj_graph.php",
+    "stats/jpgraph_files/users_logging_on.php",
+    "stats/jpgraph_files/users_by_language.php",
+    "stats/jpgraph_files/total_proj_graph.php",
+    "stats/jpgraph_files/average_hour_users_logging_on.php",
+    "stats/jpgraph_files/total_pages_by_month_graph.php",
+    "stats/jpgraph_files/new_users.php",
+    "stats/jpgraph_files/cumulative_month_proj.php",
+    "stats/jpgraph_files/tallyboard_deltas.php",
+    "stats/jpgraph_files/curr_month_proj.php",
+    "stats/jpgraph_files/equilibria.php",
+    # Design documentation
+    "styles/style_demo.php",
+    "styles/design_philosophy.php",
+    # Tools
+    "tools/setlangcookie.php",  # allows unauth users to set their language
+    "tools/post_proofers/smooth_reading.php",
+    "tools/proofers/ctrl_frame.php",  # required for unauth quiz access
+];
+
+$files = get_all_php_files("../");
+foreach($files as $file)
+{
+    # If it requires authentication, skip it
+    if(file_requires_auth("../$file"))
+        continue;
+
+    # If it's in the SETUP directory, skip it
+    if(startswith($file, "SETUP/"))
+        continue;
+
+    # If it's on the OK list, skip it
+    if(in_array($file, $ok_files))
+        continue;
+
+    echo "$file does not require authentication and is not on the known-good list\n";
+    exit(1);
```

Comment:
Maybe abort after finding all, rather than after just the first.

Reply:
All of our linting checks currently stop after the first failure, otherwise it's hard to find the ones with the errors since it outputs all of the filenames. And while we could stop outputting the names unless there was an error, we'd only output the failures in a single check script instead of all of them and then stopping. Either way, for this and the other MR it's better to be consistent with our current lint check behaviour.

Reply:
Ah, okay.